tests/tape/test_qnode.py

```diff
@@ -330,6 +330,44 @@ def func(x):
             == "<QNode: wires=1, device='default.qubit.autograd', interface='autograd', diff_method='best'>"
         )
 
+    @pytest.mark.parametrize("par", [None, 1, 1.1, np.array(1.2)])
+    def test_diff_method_none(self, par):
+        """Test if diff_method=None works as intended."""
+        dev = qml.device("default.qubit", wires=1)
+
+        def func(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        qn = qml.QNode(func, dev, diff_method=None)
+        assert qn.interface is None
+
+        grad = qml.grad(qn)
+
+        # Raise error in all cases
+        # Case 1: No input
+        # Case 2: int input
+        # Case 3: float input
+        # Case 4: numpy input
+        with pytest.raises(TypeError) as exp:
+            grad() if par is None else grad(par)
+
+    def test_diff_method_none_no_qnode_param(self):
+        """Test if diff_method=None works as intended."""
+        dev = qml.device("default.qubit", wires=1)
+
+        def func():
+            qml.PauliX(wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        qn = qml.QNode(func, dev, diff_method=None)
+        assert qn.interface is None
+
+        grad = qml.grad(qn)
+
+        # No differentiation required. No error raised.
+        grad()

```

Comment:
Hey @ankit27kh --- this is a small flyby comment, and apologies if I have already missed the conversation, but I had a thought seeing this:

- When `diff_method=None`, the output of `qml.grad(qn)(*args)` should be an array of zeros, so you could check if that is the case?
- Furthermore, in Autograd, a warning is raised (`UserWarning: Output seems independent of input.`). You could test that this occurs using `with pytest.warns(UserWarning, match="Output seems independent"):`

Reply:
Hi @josh146, what do you think about this?
```
@pytest.mark.parametrize("par", [None, [0.0, [1.1, 2.2], np.array(3)]])
def test_diff_method_none_no_qnode_param(self, par):
    """Test if diff_method=None works as intended."""
    dev = qml.device("default.qubit", wires=1)

    def func(*args):
        qml.PauliX(wires=0)
        return qml.expval(qml.PauliZ(0))

    qn = qml.QNode(func, dev, diff_method=None)
    assert qn.interface is None

    grad = qml.grad(qn)

    # No differentiation required. No error raised.
    with pytest.warns(UserWarning, match="Output seems independent"):
        if par is None:
            grad()
        else:
            assert grad(par) == [np.array(0.), [np.array(0.), np.array(0.)], np.array(0)]
```
This covers both of your points.

Reply:
Hey @ankit27kh! This looks great, very nice!

Only two suggestions:

1. there is maybe no point including the `par=None` case in your tests, since the gradient is not defined if the function has no arguments 🙂 

2. Secondly, it would be worth having the arguments be used within the QNode. If they are not used, then the gradient will always be zero, even when `diff_method != None`!

e.g.,
```python

    def func(*args):
        qml.PauliX(wires=0)
        qml.RX(args[0], wires=0)
        return qml.expval(qml.PauliZ(0))
```

Reply:
@josh146 
>no point including the par=None case in your tests, since the gradient is not defined

Yes, there are no arguments. But we are also specifically checking that case to make sure there are no errors.

> it would be worth having the arguments be used within the QNode

We have another test where we use the input. This case is for the situation where we don't use the input.
'diff_method=None` will error out if you try to use grad with inputs. 

The whole point was that if you have set `diff_method=None` then don't expect grads!

Reply:
Oh, got it :) Then my comments can be ported to the other test, if applicable. If already taken into account, feel free to ignore!